csrc/id_model/id_model.h

```diff
@@ -0,0 +1,137 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <disjoint_set.h>
+#include <fusion.h>
+#include <ir/all_nodes.h>
+#include <val_graph.h>
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace nvfuser {
+
+class ValGraph;
+
+// A collection of IterDomainGraphs that are built from a fusion or series of
+// expressions. These graphs are related, but have some distinct features based
+// on the IdMappingMode.
+//
+// EXACT/PERMISSIVE mode:
+//
+// consumer[i0, b1] = producer[i0]
+// consumer->merge(0) (consumer will now be [i0 * b1])
+//
+// When producer is replayed as consumer (the direction we use for mapping)
+// with forwarding from ForwardingInfo the producer to consumer map will have
+// both a mapping of consumer(i0) to producer(i0) as well as consumer(i0*b1) to
+// producer(i0). This latter mapping is important for loop nest mappings as the
+// consumer will generate a loop based on i0*b1 and the producer may be
+// computeAt inside this loop nest. However, for indexing we do not want these
+// two iter domains mapped as producer may be indexed as i0*i1 depending on the
+// loop nest structure and how it was built.
+//
+// Exact mode is if the iter domain relationships from producer to consumer are
+// considered the exact same size operating on matching dimensions from the root
+// domain mapping.
+//
+// IdMappingMode::EXACT
+//   Don't map any broadcast axes to non-broadcast axes
+//   Do not forward through any broadcast IDs
+//
+class IdModel : public PolymorphicBase {
+ public:
+  IdModel(
+      const std::vector<Expr*>& exprs,
+      const std::vector<TensorView*>& additional_tvs,
```

Comment:
I didn't find this PR construct an `IdModel` with `additional_tvs`. Is this still useful? If so, under what situations?

Reply:
This is used from `IdModel::IdModle(Fusion*)`. It's just to make sure all input and output tensors are analyzed even if they may not used, like aliased tensors.